session.go

```diff
@@ -774,8 +784,9 @@ func (s *session) tryQueueingUndecryptablePacket(p *receivedPacket) {
 		return
 	}
 	utils.Infof("Queueing packet 0x%x for later decryption", p.publicHeader.PacketNumber)
-	if len(s.undecryptablePackets)+1 >= protocol.MaxUndecryptablePackets {
-		s.close(qerr.Error(qerr.DecryptionFailure, "too many undecryptable packets received"))
+	if len(s.undecryptablePackets)+1 >= protocol.MaxUndecryptablePackets && s.receivedTooManyUndecrytablePackets.IsZero() {
+		s.receivedTooManyUndecrytablePackets = time.Now()
+		s.maybeResetTimer()
```

Comment:
I think we should return here and not append the packet to the list, so that the size is always constrained by MaxUndecryptablePackets.